src/views/smart_table.rs

```diff
@@ -0,0 +1,219 @@
+use vizia::prelude::*;
+
+pub const CELL_MIN_SIZE_PX: f32 = 100.0;
+
+#[derive(Clone, Copy, Debug)]
+pub enum SmartTableEvent {
+    StartDrag(usize),
+    StopDrag,
+}
+
+#[derive(Lens)]
+pub struct SmartTable {
+    dragging: Option<usize>,
+    initialized: bool,
+    limiters: Vec<f32>,
+    sizes: Vec<f32>, // derived
+}
+
+impl<'a> SmartTable {
+    pub fn new<S>(cx: &'a mut Context, mut rows: Vec<Vec<S>>) -> Handle<'a, Self>
+    where
+        S: Into<&'a str> + Clone,
```

Comment:
Maybe this should be `ToString` instead so that it doesn't require a lifetime?

Reply:
Ah, true. I'll make that change ASAP